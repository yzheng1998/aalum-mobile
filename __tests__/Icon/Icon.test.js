import React from 'react'
import Icon from '../../src/components/Icon'
import { shallow } from 'enzyme'

describe('Icon', () => {
  test('renders correctly on iOS', () => {
    const wrapper = shallow(<Icon name="heart" />)
    expect(wrapper).toMatchSnapshot()
  })

  test('renders correctly on Android', () => {
    jest.mock('Platform', () => {
      const Platform = jest.requireActual('Platform')
      Platform.OS = 'android'
      return Platform
    })
    const wrapper = shallow(<Icon name="heart" />)
    expect(wrapper).toMatchSnapshot()
  })
})
