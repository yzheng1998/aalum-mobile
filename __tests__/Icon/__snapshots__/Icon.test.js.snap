// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly on Android 1`] = `
<Icon
  allowFontScaling={false}
  name="md-heart"
  size={12}
/>
`;

exports[`Icon renders correctly on iOS 1`] = `
<Icon
  allowFontScaling={false}
  name="ios-heart"
  size={12}
/>
`;
